plugins/network-elements/nuage-vsp/src/com/cloud/network/manager/NuageVspManagerImpl.java

```diff
@@ -285,60 +507,341 @@ public boolean deleteNuageVspDevice(DeleteNuageVspDeviceCmd cmd) {
+        // Whenever a Nuage VSP Host comes up, check if all CS domains are present and check if the CMS ID is valid
+        if (transition.getToState() == Status.Up && vo instanceof HostVO) {
+            auditHost((HostVO) vo);
         }
+        return true;
+    }
+
+    @Override
+    public boolean configure(String name, Map<String, Object> params) throws ConfigurationException {
+        initMessageBusListeners();
+        initNuageVspResourceListeners();
+        initNuageNetworkOffering();
+        initNuageVspVpcOffering();
+        Status.getStateMachine().registerListener(this);
+        return true;
+    }
+
+    @DB
+    private void initMessageBusListeners() {
+        // Create corresponding enterprise and profile in VSP when creating a CS Domain
+        _messageBus.subscribe(DomainManager.MESSAGE_ADD_DOMAIN_EVENT, new MessageSubscriber() {
+            @Override
+            public void onPublishMessage(String senderAddress, String subject, Object args) {
+                Long domainId = (Long) args;
+                Domain domain = _domainDao.findById(domainId);
+
+                try {
+                    _domainDao.acquireInLockTable(domain.getId());
+
+                    List<NuageVspDeviceVO> nuageVspDevices = _nuageVspDao.listAll();
+                    for (NuageVspDeviceVO nuageVspDevice : nuageVspDevices) {
+                        HostVO host = findNuageVspHost(nuageVspDevice.getHostId());
+                        SyncDomainCommand cmd = new SyncDomainCommand(domain.getUuid(), domain.getName(), domain.getPath(), true, false);
+                        _agentMgr.easySend(host.getId(), cmd);
+                    }
+                } catch (Exception e) {
+                    s_logger.error(e.getMessage());
+                } finally {
+                    _domainDao.releaseFromLockTable(domain.getId());
+                }
+            }
+        });
+
+        // Delete corresponding enterprise and profile in VSP when deleting a CS Domain
+        _messageBus.subscribe(DomainManager.MESSAGE_REMOVE_DOMAIN_EVENT, new MessageSubscriber() {
+            @Override
+            public void onPublishMessage(String senderAddress, String subject, Object args) {
+                DomainVO domain = (DomainVO) args;
+                try {
+                    List<NuageVspDeviceVO> nuageVspDevices = _nuageVspDao.listAll();
+                    for (NuageVspDeviceVO nuageVspDevice : nuageVspDevices) {
+                        HostVO host = findNuageVspHost(nuageVspDevice.getHostId());
+                        SyncDomainCommand cmd = new SyncDomainCommand(domain.getUuid(), domain.getName(), domain.getPath(), false, true);
+                        _agentMgr.easySend(host.getId(), cmd);
+                    }
+                } catch (Exception e) {
+                    s_logger.error(e.getMessage());
```

Comment:
Pass the exception instance into the logger to capture the full stack trace for debugging.